### DISCUSSION TITLE
Textadept 11.4 alpha

### DISCUSSION THREAD

**Discussion by orbitalquark** [orbitalquark/textadept#193] [General]
Textadept 11.4 alpha is released from https://github.com/orbitalquark/textadept/releases/tag/textadept_11.4_alpha

This is a big new alpha release with the following notable changes:

* GTK 3 is used instead of GTK 2 by default (GTK is the cross-platform GUI toolkit that Textadept uses).
* Windows binaries are now 64-bit. 32-bit support has been dropped.
* Mac OSX 10.10 (Yosemite) is the new minimum version required. (This is a side-effect of switching to GTK 3.)
* Tabs are rearrangeable via drag and drop.
* Makefile rules have changed a bit, and the GUI and terminal versions can be compiled concurrently.

More information and links are available below.

Download:

* [Textadept 11.4 alpha  -- Windows][]
* [Textadept 11.4 alpha  -- Mac OSX 10.10+][]
* [Textadept 11.4 alpha  -- Linux][]
* [Textadept 11.4 alpha  -- Modules][]

Bugfixes:

* Fixed bug in `ui.goto_view()` when specifying a preferred view.
* Fixed busy loop in `ui.update()` on macOS when monitoring output of spawned processes.
* Ensure preferred "lexer.trigger.ext" snippet files are used over "trigger.ext".
* Improved Windows network directory path handling.
* Fixed alpha color value of brace matches.
* Fixed busy wait on spawned process exit in some cases.
* Resize line number margin on reset.
* Fixed search wrapping indicator not showing up in the statusbar.
* Fixed "Find Next" for zero-width regex searches.
* Fixed calling `view:set_theme()` from the command entry.
* Scintilla: Fixed assertion failure with autocompletion lists when order is `ORDER_CUSTOM` or `ORDER_PERFORMSORT`.
* Scintilla: On Wayland, fixed autocompletion window display on secondary monitor.
* Scintilla: Fixed some scrollbar inaccuracies with annotations and wrapped lines.
* Scintilla: Improved Chinese character alignment with roman characters.

Changes:

* Use [GTK][] 3 by default on all platforms.
* Windows binaries are 64-bit now; dropped support for 32-bit Windows.
* The minimum required version of macOS is 10.10 (Yosemite).
* Tabs are now rearrangeable via drag and drop.
* Makefile allows building both the GUI and terminal versions concurrently.
* Updated test suite to run on Windows and macOS.
* Use Free Desktop Icon Naming Specification icon names in dialogs.
* New [Docker image][] for building with GTK 3.
* Added [`ui.find.show_filenames_in_progressbar`][] option for hiding filenames during Find in
Files searches.
* Added [`move_buffer()`][] function for rearranging buffers.
* Added support for flow control sequences in the non-Windows terminal version with `-p` and
`--preserve` command line options.
* Updated Fennel lexer.
* Updated Python lexer to highlight class definitions.
* Scintilla: Improve performance for very long lines.
* Scintilla: Shift+Mouse Wheel scrolls horizontally.
* Updated to [Scintilla][] 5.2.0.
* Updated to [Lua][] 5.4.4.
* Updated to libtermkey 0.22.

[Textadept 11.4 alpha  -- Windows]: https://github.com/orbitalquark/textadept/releases/download/textadept_11.4_alpha/textadept_11.4_alpha.win.zip
[Textadept 11.4 alpha  -- Mac OSX 10.10+]: https://github.com/orbitalquark/textadept/releases/download/textadept_11.4_alpha/textadept_11.4_alpha.macOS.zip
[Textadept 11.4 alpha  -- Linux]: https://github.com/orbitalquark/textadept/releases/download/textadept_11.4_alpha/textadept_11.4_alpha.linux.tgz
[Textadept 11.4 alpha  -- Modules]: https://github.com/orbitalquark/textadept/releases/download/textadept_11.4_alpha/textadept_11.4_alpha.modules.zip
[GTK]: https://gtk.org
[Docker image]: https://orbitalquark.github.io/textadept/manual.html#compiling-using-docker
[`ui.find.show_filenames_in_progressbar`]: https://orbitalquark.github.io/textadept/api.html#ui.find.show_filenames_in_progressbar
[`move_buffer()`]: https://orbitalquark.github.io/textadept/api.html#move_buffer
[Scintilla]: https://scintilla.org
[Lua]: https://lua.org

**Comment by amitbha** [comment]
Version: [textadept_11.4_alpha](https://github.com/orbitalquark/textadept/releases/download/textadept_11.4_alpha/textadept_11.4_alpha.linux.tgz)
OS: Manjaro 21.2.5

Tabs are still not dragable.

Update: 
Got it. The left and right edges of tab is dragable!

**Comment by orbitalquark** [comment]
From the manual: Tabs can be rearranged by clicking on their tops and dragging and dropping them.

It's not ideal, but at the moment you have to click above the tab text to drag.

**Reply by amitbha** [reply]
tops work occasionally.
sides work for me.
![ksnip_20220404-150403](https://user-images.githubusercontent.com/4060625/161491197-0f3dd214-17c5-4781-a10a-9ee69d3028c6.png)

**Comment by brynne8** [comment]
Textadept requires new version of glibc package that Debian 10 doesn't have.
```
./textadept: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by ./textadept)
./textadept: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by ./textadept)
```

**Comment by orbitalquark** [comment]
That's unfortunate. The manual mentions that this could happen on Linux systems. You'll have to compile Textadept manually[1] on your machine. Binaries are built using an Ubuntu 20.04 image. Debian 10 was first released in 2019, so that's probably why there's a libc and libstdc++ mismatch.

[1]: https://orbitalquark.github.io/textadept/manual.html#compiling
